service/src/main/kotlin/fi/espoo/evaka/varda/VardaOrganizerChild.kt

```diff
@@ -0,0 +1,209 @@
+package fi.espoo.evaka.varda
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.annotation.JsonProperty
+import fi.espoo.evaka.shared.db.Database
+import fi.espoo.evaka.varda.integration.VardaClient
+import fi.espoo.evaka.varda.integration.convertToVardaRequest
+import mu.KotlinLogging
+import org.jdbi.v3.core.kotlin.mapTo
+import java.util.UUID
+
+private val logger = KotlinLogging.logger {}
+
+fun getOrCreateVardaChildByOrganizer(
+    db: Database.Connection,
+    client: VardaClient,
+    evakaPersonId: UUID,
+    organizerOid: String,
+    sourceSystem: String
+): Long {
+    return db.transaction { tx ->
+        val municipalOrganizerOid = getMunicipalOrganizerOid(tx)
+        val isPaosChild = organizerOid != municipalOrganizerOid
+
+        val rowsByChild = getVardaOrganizationChildRows(tx, evakaPersonId)
+        if (rowsByChild.isEmpty()) {
+            return@transaction createVardaPersonAndChild(tx, client, evakaPersonId, municipalOrganizerOid, organizerOid, isPaosChild, sourceSystem)
+        }
+
+        val vardaPersonOid = rowsByChild.first().vardaPersonOid
+
+        val rowsByChildAndOrganizer = rowsByChild.filter { row -> row.organizerOid == organizerOid }
+        if (rowsByChildAndOrganizer.isEmpty()) {
+            return@transaction createVardaChildWhenPersonExists(tx, client, evakaPersonId, vardaPersonOid, municipalOrganizerOid, organizerOid, isPaosChild, sourceSystem)
+        }
+
+        return@transaction rowsByChildAndOrganizer.first().vardaChildId
+    }
+}
+
+data class VardaChildOrganizerRow(
+    val evakaPersonId: UUID,
+    val vardaPersonOid: String,
+    val vardaChildId: Long,
+    val organizerOid: String
+)
+
+private fun getVardaOrganizationChildRows(
+    tx: Database.Transaction,
+    evakaPersonId: UUID,
+): List<VardaChildOrganizerRow> {
+    val sql = """
+        SELECT evaka_person_id, varda_person_oid, varda_child_id, organizer_oid
+        FROM varda_organizer_child
+        WHERE evaka_person_id = :evakaPersonId
+    """.trimIndent()
+
+    return tx.createQuery(sql)
+        .bind("evakaPersonId", evakaPersonId)
+        .mapTo<VardaChildOrganizerRow>()
+        .toList()
+}
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+data class VardaChildPayload(
+    @JsonProperty("henkilo_oid")
+    val personOid: String,
+    @JsonProperty("vakatoimija_oid")
+    val organizerOid: String,
+    @JsonProperty("lahdejarjestelma")
+    val sourceSystem: String
+)
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+data class VardaPaosChildPayload(
+    @JsonProperty("henkilo_oid")
+    val personOid: String,
+    @JsonProperty("oma_organisaatio_oid")
+    val organizerOid: String,
+    @JsonProperty("paos_organisaatio_oid")
+    val paosOrganizationOid: String,
+    @JsonProperty("lahdejarjestelma")
+    val sourceSystem: String
+)
+
+private fun getMunicipalOrganizerOid(tx: Database.Transaction): String {
```

Comment:
Does this ever change? Should be moved to a property (no need to do it in this PR)

Reply:
This is like the first question that ever appeared with Varda integration :D The answer remains... Let's do it later.